### DISCUSSION TITLE
SDK++: ideas for distributing a more complete developer environment

### DISCUSSION THREAD

**Discussion by beriberikix** [zephyrproject-rtos/zephyr#43211] [Ideas]
The next generation SDK (#37255) is nearly [released ](https://github.com/zephyrproject-rtos/sdk-ng-testing/releases)and will finally make a available cross-platform SDK for the major 3 targets. The scope of the current SDK explicitly limited to toolchains and host tools. There has been various discussions of what a more complete developer environment might include (I'm calling it SDK++ for now.)

To start us off, I'll list out some of the ideas that was discussed in the last ZDS, in a previous [general discussion about DevExp](https://github.com/zephyrproject-rtos/zephyr/discussions/36194) and my own ideas:

- Distribute SDKs via package managers (Ex. brew install zephyr)
- Distribute common dependencies (Ex. CMake)
- Distribute explicit or OS-level dependencies (Ex. Git)
- Create native installers (ex. MSIX, .deb)
- Distribute configuration metadata for different Editors/IDEs (ex. VSCode setting.json, devcontainer.json; Emacs project settings)
- Create native extensions for different Editors/IDEs
  - Distribute through IDEs where possible (ex. VSCode Marketplace)
- Integration with or configuration metadata for CI/CD systems (Ex. Jenkins, Github Actions)
- Visual configuration tools (ex. STM32CubeMX)
- Improvements to container support (ex. smaller images, platform-specific images)

Each bullet is a discussion topic on their own but we'd love to gather additional suggestions, or simply gather feedback on any one!

**Comment by hongshui3000** [comment]
>Create native extensions for different Editors/IDEs
Distribute through IDEs where possible (ex. VSCode Marketplace)

I would love to have vscode extension distribution, which will make it easier for me to use vscode for zephyrrtos related development
